### PR TITLE
Releaseable point cloud

### PR DESCRIPTION
**Pull request by trym-b**
Instead of transforming the point cloud directly to a numpy buffer, we
should instead return a intermediate layer, since in the underlying C++
API point cloud exposes various functions such as width and height.

Making buffer releasable also created some technical debt:
assertNotReleased is needed since accessing the underlying a C++ object
after it has been released causes a crash.

### CODE REVIEW

File: modules/zivid/point_cloud.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        Returns:
+            a numpy array
+
+        """
+        self.__impl.assert_not_released()
+        return numpy.array(self.__impl)
+
+    @property
+    def height(self):
+        """Maximum height of point cloud.
```
Comment (nedrebo) [inline modules/zivid/point_cloud.py]: When 1:1 wrapping a function from C++ just copy the documentation.

 * Consistency
 * Time already spent on thinking about wording. No point in spending that time again.

This goes for all functions in PR and (also for the all functions in the entire wrapper, but that is not part of this PR).

Comment (trym-b) [reply modules/zivid/point_cloud.py]: Fixed.

**Comment by trym-b** [comment]
Ready for next round.

File: modules/zivid/point_cloud.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            a positive integer
+
+        """
+        return self.__impl.width()
+
+    def release(self):
+        """Release the underlying resources."""
+        self.__impl.release()
+
+    def __enter__(self):
```
Comment (nedrebo) [inline modules/zivid/point_cloud.py]: Maybe we should have a mixin or base class or something for these three functions. They are duplicated many places now.

You are just adding to the existing solution, so no need to fix in this PR. But if you agree please add an issue for it.

Comment (trym-b) [reply modules/zivid/point_cloud.py]: Could probably be done through a decorator:
```
@utils.context_manager
class SomethingThatRequiresContextManager:
    ....
```
Will create an issue

Comment (trym-b) [reply modules/zivid/point_cloud.py]: https://github.com/zivid/zivid-python/issues/27
